compiler/src/iree/compiler/Codegen/Common/TileDispatchUsingInterface.cpp

```diff
@@ -339,12 +324,26 @@ tileDispatchUsingSCFFopOp(RewriterBase &rewriter, TilingInterface op,
       // The parallel loops that are tiled are partitionable loops.
       SmallVector<Range> parallelLoopRanges;
       SmallVector<unsigned> partitionedLoopIds;
+
+      AffineExpr s0, s1, s2, s3; // lb, ub, step, tileSize
+      bindSymbols(rewriter.getContext(), s0, s1, s2, s3);
+      AffineExpr numTilesExprs = (s1 - s0).ceilDiv(s2 * s3);
       for (auto [index, iteratorType] : llvm::enumerate(iteratorTypes)) {
-        if (iteratorType == utils::IteratorType::parallel &&
-            !isZero(tileSizeVector[index])) {
-          parallelLoopRanges.push_back(iterationDomainOfr[index]);
-          partitionedLoopIds.push_back(index);
+        if (iteratorType != utils::IteratorType::parallel ||
+            isConstantIntValue(tileSizes[index], 0)) {
+          continue;
         }
+
+        OpFoldResult numTiles = affine::makeComposedFoldedAffineApply(
+            rewriter, loc, numTilesExprs,
+            {iterationDomain[index].offset, iterationDomain[index].size,
```

Comment:
just curious is iterationDomain[index].size the UB / last value of the loop's IV? 

if it is then it makes sense if not, not really obvious to me how it this computation is numTiles.

Reply:
I stared at this for a while too.. Here is my reasoning

The original loop is something like 

```
scf.for %iv = %lb to %ub step %step
```

Tiling by `%tile_size` can be represented as

```
scf.for %iv0 = %lb to %ub step %step * %tile_size
  scf.for %iv = %iv0 to min(%iv0 + %step * %tile_size, %ub) step %step
```

The number of tiles is the number of iterations of the outer loop  which is `(%ub - %lb).ceilDiv(%step * %tile_size)`?

Reply:
I see, that makes sense, thanks!